gddo-server/main.go

```diff
@@ -760,12 +761,30 @@ func handleAPIError(resp http.ResponseWriter, req *http.Request, status int, err
 	json.NewEncoder(resp).Encode(&data)
 }
 
-type hostMux []struct {
+type rootHandler []struct {
 	prefix string
 	h      http.Handler
 }
 
-func (m hostMux) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
+func (m rootHandler) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
+	host := req.Host
+	if h, _, err := net.SplitHostPort(host); err == nil {
+		host = h
+	}
+	if host == "godoc.org" {
+		// Because https is not used api.godoc.org, the includeSubDomains
+		// parameter is not used here.
+		resp.Header().Add("Strict-Transport-Security", "max-age=631138519; preload")
```

Comment:
> Hey @agl does this header look right to you? I'm concerned about the length of the timeout. What's the best practice?

The timeout should be at least 18 weeks (10886400). That value is 20
years, which is effectively infinity. It's certainly greater than 18
weeks ;)